styx-api-service/src/main/java/com/spotify/styx/api/workflow/WorkflowInitializer.java

```diff
@@ -50,7 +50,7 @@ public WorkflowInitializer(Storage storage, Time time) {
   public Optional<Workflow> store(Workflow workflow, Consumer<Optional<Workflow>> guard)
       throws WorkflowInitializationException {
     try {
-      return storage.runInTransaction(tx -> store(tx, workflow, guard));
+      return storage.runInTransactionWithRetries(tx -> store(tx, workflow, guard));
```

Comment:
Not sure we want to retry on conflict here. We might end up writing stale data. If there is another transaction won, this one can give up.

Reply:
Same